File: back/DevC/src/main/java/com/back/devc/domain/post/post/dto/AdminPostListResponse.java
package com.back.devc.domain.post.post.dto;

import com.back.devc.domain.post.post.entity.Post;

import java.time.LocalDateTime;

//게시글 전체 조회로 간략한 정보만 담아서 전달
public record AdminPostListResponse(
        Long postId,
        String title,
        Long userId,
        Long categoryId,
        Boolean isDeleted,
        int viewCount,
        int likeCount,
        int commentCount,
        LocalDateTime createdAt
) {
    public AdminPostListResponse(Post post) {
        this(
                post.getPostId(),
                post.getTitle(),
                post.getMember() != null ? post.getMember().getUserId() : null,
                post.getCategory().getCategoryId(),
                post.isDeleted(),
                post.getViewCount(),
                post.getLikeCount(),
                post.getCommentCount(),
                post.getCreatedAt()
        );
    }
}

File: back/DevC/src/main/kotlin/com/back/devc/domain/post/post/dto/AdminPostListResponse.kt
package com.back.devc.domain.post.post.dto

import com.back.devc.domain.post.post.entity.Post
import java.time.LocalDateTime

// 게시글 전체 조회용 간략 정보 DTO
data class AdminPostListResponse(
    val postId: Long,
    val title: String,
    val userId: Long?,
    val categoryId: Long,
    val isDeleted: Boolean,
    val viewCount: Int,
    val likeCount: Int,
    val commentCount: Int,
    val createdAt: LocalDateTime
) {
    companion object {
        fun from(post: Post): AdminPostListResponse {
            return AdminPostListResponse(
                postId = post.postId,
                title = post.title,
                userId = post.member?.userId,
                categoryId = post.category.categoryId,
                isDeleted = post.isDeleted,
                viewCount = post.viewCount,
                likeCount = post.likeCount,
                commentCount = post.commentCount,
                createdAt = post.createdAt
            )
        }
    }
}